ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -0,0 +1,163 @@
+ *                      the fetcher will still read and send all data in June 2013, as if end was set to 2013-06-30 24:00 
+ * @param speed         The speed at which the fetcher will fetch quotes. Defaults to 1 (which means quotes are replayed
+ *                      as they were recorded). Can be set e.g. to 2 (time runs twice as fast) or 60 (one hour of historical
+ *                      quotes is sent in one minute), etc. Time steps are dT = int(1/speed * dT_historical), in milliseconds.
+ *                      Accordingly: be careful with high speeds, if e.g. speed=1000 the system can't make the difference
+ *                      between dT_historical=4200 and dT_historical=4900, they will be both sent after 4ms in the simulation.
+ *                      If we go even higher, dT will become < 1 and thus 0, which in theory would mean send all the quotes
+ *                      you read at the same time, but in practice would probably f*ck things.
+ */
+
+class HistDataCSVFetcher(dataDir: String, 
+                         currencyPair: String, 
+                         start: Date, 
+                         end: Date,
+                         speed: Double = 1.0
+                        ) 
+                         extends PushFetchComponent[Quote] {
+  
+  val workingDir = dataDir+"/"+currencyPair.toUpperCase()+"/";
+  val (whatC, withC) = Currency.pairFromString(currencyPair);
+  
+  val bidPref = "DAT_NT_"+currencyPair.toUpperCase()+"_T_BID_"
+  val askPref = "DAT_NT_"+currencyPair.toUpperCase()+"_T_ASK_"
+  
+  /**
+   * The centerpiece of this class, where we actually load the data.
+   * It contains all quotes this fetcher reads from disk, ready to be fetch()ed
+   */
+  val allQuotes: List[Quote] = monthsBetweenStartAndEnd.flatMap(m => parse(bidPref+m+".csv", askPref+m+".csv")).sortBy { q => q.timestamp }
+  
+  /**
+   * Index of the next quote to be fetched, incremented whenever a quote has been fetched
+   */
+  var quoteIndex = 0
+  
+  /**
+   * Using java.util.Timer to simulate the timing of the quotes when they were generated originally.
+   * Schedules a new version of itself t milliseconds after it has send the current quote,
+   * 
+   * t = (time when next quote was recorded) - (time when current quote was recorded)
+   */
+  val timer = new Timer()
+  timer.schedule(new SendQuotes, 0)
+  class SendQuotes extends java.util.TimerTask {
+    def run() {
+      if (quoteIndex < allQuotes.length) {
+        //update the iterator variables
+        var currQ = allQuotes( quoteIndex )
+        var nextQ = allQuotes( List(quoteIndex+1,allQuotes.length).min )
+        quoteIndex = quoteIndex + 1;
+        
+        //send the quote and schedule next call
+        callback(currQ)
+        timer.schedule(new SendQuotes(), (1/speed*(nextQ.timestamp - currQ.timestamp)).toInt)
+      } else {
+        timer.cancel();
+      }
+    }
+  }
+  
+  //TODO
+  def loadInPersistor(filename: String) {  }
```

Comment:
I'll remove it completely for now, it's wrong argument type anyways. I will add it in the quotepersistor branch.